src/Listener/TerminateListener.php

```diff
@@ -165,4 +179,58 @@ private function validateInputMap(array $inputMap, string $commandClass): void
             Assert::allString($value, 'Malformed input mapper for ' . $commandClass);
         }
     }
+
+    /**
+     * @psalm-return non-empty-string
+     */
+    private function getVendorDirectory(): string
+    {
+        $basePath     = realpath(getcwd());
+        $composerJson = file_get_contents($basePath . '/composer.json');
+        Assert::string($composerJson);
+
+        $composer = json_decode($composerJson, true);
+        Assert::isMap($composer);
+
+        $vendorDir = $composer['config']['vendor-dir'] ?? $basePath . '/vendor';
+        Assert::string($vendorDir);
+        Assert::directory($vendorDir);
```

Comment:
this might be extreme case, but possible according to composer documentation:

> `$HOME` and `~` will be replaced by your home directory's path

https://getcomposer.org/doc/06-config.md#vendor-dir